docs/local_dev_environment.md

```diff
@@ -0,0 +1,219 @@
+# Set up a local development environment
```

Comment:
I believe the current `indexd` `README.md` has some instructions already. I think you've made it so that this file covers almost everything in the `README.md`, but can you delete the now-redundant `README.md` info and link to this file from there?

Reply:
Updated and thanks for the feedback.  Let me know if this works.